### PR TITLE
Create Gather IFP Data action

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
- Create `Ifp` class script include
- Create `DOCGEN: Gather IFP Data` action
```javascript
  var ifp = new Ifp(inputs.package);
  outputs.payload = JSON.stringify(ifp.getPayload());
```

Ticket: AT-8378

### CODE REVIEW

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_5070b8dc976fd1106fa8b4b3f153afdd.xml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+          gtcNumber: fundingRequest.fs_form.gtc_number.toString(),
+          orderNumber: fundingRequest.fs_form.order_number.toString(),
+        };
+
+      } catch (err) {
+        throw this.errUtil.createError("IFP --> getFundingDocument() " + err, this.errUtil.METHOD_ERROR);
+      }
+    },
+    /**
+     * Lookups up all incremental funding records from a Funding Plan
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_5070b8dc976fd1106fa8b4b3f153afdd.xml:86]:
```suggestion
     * Looks up all incremental funding records from a Funding Plan
```

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_5070b8dc976fd1106fa8b4b3f153afdd.xml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            requirementsTitle: acqPackage.project_overview.title.toString(),
+            missionOwner: moList[0].formalName, // first mission owner
+            estimatedTotalAmount: parseFloat(fundingPlan.estimated_task_order_value.toString()),
+            initialAmount: parseFloat(fundingPlan.initial_amount.toString()),
+            remainingAmount: parseFloat(fundingPlan.remaining_amount.toString()),
+            fundingDocument: this.getFundingDocument(fundingRequest),
+            requirementAmountIncrements: this.getFundingIncrements(fundingPlan),
+            scheduleText: fundingPlan.schedule_text.toString(),
+            contractNumber: "TBD",
+            taskOrderNumber: "TBD",
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_5070b8dc976fd1106fa8b4b3f153afdd.xml:155]: thumbs up - provided by DITCO

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_97c87b02974611106fa8b4b3f153afef.xml [hunk 3]
```diff
@@ -65,6 +66,9 @@ ErrorHandler.prototype = {
 			case this.COST_RESPONSE_ERROR:
 				gs.error(error.message);
 				break;
+			case this.METHOD_ERROR:
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_97c87b02974611106fa8b4b3f153afef.xml:69]: @juliusfitzhugh-ccpo Can you document for posterity the intended use of this new error type?

**Review by jasonburkert-ccpo — APPROVED** [review]
Minor suggestions, looks good.

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_97c87b02974611106fa8b4b3f153afef.xml (hunk 3 above)
Comment (juliusfitzhugh-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_97c87b02974611106fa8b4b3f153afef.xml:83]: Added a comment above `this.METHOD_ERROR`. See 842488d532983e678481c2eb5b32e9576e82ef6a.

**Pull request merged**